src/components/Calendar/RangeCalendar/GenericRangeCalendar/GenericRangeCalendar.tsx

```diff
@@ -28,24 +32,62 @@ export function GenericRangeCalendar({
     return isInTheHoverRange(day, hoverDate)
   }
 
-  const onMouseLeave = (e: MouseEvent<HTMLDivElement>) => setHoverDate(undefined)
-  return (
-    <Calendar
-      {...rest}
-      onMouseLeave={onMouseLeave}
-      onDayHover={setHoverDate}
-      isDaySelected={isInTheRange}
-      modifiers={{
-        ...rest.modifiers,
-        inTheHoverRange: inHoverRange,
-        inTheRange: inRange,
-      }}
-      modifierStyles={{
-        ...rest.modifierStyles,
-        inTheHoverRange: dayHoverStyle,
-        inTheRange: defaultModifierStyles.selected,
-      }}
-    />
-  )
+  const inWeekRange = (week: Week): boolean => {
+    return isInTheRange(week.start) && isInTheRange(week.end)
+  }
+
+  const inHoverWeekRange = (week: Week): boolean => {
+    return hoverWeek && isInTheHoverRange(week.start, hoverWeek.start) && isInTheHoverRange(week.end, hoverWeek.end)
+  }
+
+  const onMouseLeave = (e: MouseEvent<HTMLDivElement>) => {
+    if (onlyWeeks && onlyWeeks === true) {
```

Comment:
```suggestion
    if (onlyWeeks) {
```

Isn't that redundant? Since they both check if it's true.

Reply:
I saw that the change suggestion was not made. Will it be applied?

Reply:
Está aplicada a sugestão ;)